### PR TITLE
Add ASSERT_NEQ, ASSERT_GT, ASSERT_LTE, ASSERT_GTE

### LINKED ISSUE #1: Implement assertions
The following assertions should be implemented in `assertions.h`:
 - [X] equality of integer values (`ASSERT_EQ`)
 - [x] not equal for integer values (`ASSERT_NE`)
 - [X] less than for integer values (`ASSERT_LT`)
 - [x] greater than for integer values (`ASSERT_GT`)
 - [x] less than or equal for integer values (`ASSERT_LTE`)
 - [x] greater than or equal for integer values (`ASSERT_GTE`)
 - [ ] *is 0* or *was successful* (`ASSERT_ZERO`, `ASSERT_SUCCESS`); the successful status comes from the philosophy of Unix syscalls which return 0 on success
 - [ ] string relations (`ASSERT_STRCMP(a, b, expected_relation)`)
 - [ ] equality for floating point values (`ASSERT_FLOAT_EQ`)

### LINKED ISSUE #2: Allow parsing command line arguments for running custom sets of tests
The library should provide a `main` function for the test executables.

The program should accept the following command line arguments:

 - [x] `--all` or `-a` to run all the registered tests 
 - [x] `--suite <suite_name>` or `-s <suite_name>` to run all the tests registered within the given `suite_name`
 - [ ] `--failed` to skip running the tests that failed on the last run

### PR DESCRIPTION
**Pull request by alexgrusu**
This PR closes #1 and closes #2.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "files.associations": {
+    "assertions.h": "c",
+    "ctest.h": "c"
+  }
+}
```
Comment (barbu110) [inline .vscode/settings.json]: Please remove this file as its not something that should be committed to the repository.

File: include/ctest/assertions.h [hunk 2]
```diff
@@ -4,6 +4,16 @@
 
 #pragma once
 
+/* We define these the same for all test suites. */
+#define	ASSERT_FAILURE	1	/* Failing assert status. */
+#define	ASSERT_SUCCESS	0	/* Successful assert status. */
```
Comment (barbu110) [inline include/ctest/assertions.h]: How about using an `enum` here?

File: include/ctest/ctest.h [hunk 3]
```diff
@@ -38,6 +38,8 @@ struct assertions_container
   struct assertion assertions[MAX_ASSERTIONS_PER_TEST];
 };
 
+typedef enum {false, true} bool;
```
Comment (barbu110) [inline include/ctest/ctest.h]: I am not sure this is used anywhere in the program. I know I created it initially but is it used anywhere right now?

**Review by barbu110 — CHANGES_REQUESTED** [review]

File: include/ctest/assertions.h [hunk 4]
```diff
@@ -15,3 +43,29 @@
...
+  do {                                                                   \
+    struct assertion __assertion = {">=", #lhs, #rhs, ((lhs) >= (rhs))}; \
+    cont->assertions[cont->size++] = __assertion;                        \
+  } while (0)
+
+#define ASSERT_STRCMP(lhs, rhs, EXPECTED_RELATION)                        \
+  do {                                                                    \
+    char result = strcmp(lhs, rhs);                                       \
+    char var = opi(result, EXPECTED_RELATION, 0);                         \
+    struct  assertion __assertion = {"", #lhs, #rhs, var};                \
```
Comment (barbu110) [inline include/ctest/assertions.h]: Lets use something like:

```c
__assertion = {"!=", "strcmp(" #lhs ", " #rhs ")", #EXPECTED_RELATION, var};
```

Also, there are two spaces between `struct` and `assertion`.

File: .vscode/settings.json (hunk 1 above)
Comment (alexgrusu) [reply .vscode/settings.json]: Removed.

File: include/ctest/assertions.h (hunk 2 above)
Comment (alexgrusu) [reply include/ctest/assertions.h]: Implemented as suggested.

File: include/ctest/ctest.h (hunk 3 above)
Comment (alexgrusu) [reply include/ctest/ctest.h]: Removed. Used for other tests and not necessary anymore.

File: include/ctest/assertions.h (hunk 4 above)
Comment (alexgrusu) [reply include/ctest/assertions.h]: Implemented.

**Comment by alexgrusu** [comment]
Resolves Issue #1, without ASSERT_FLOAT_EQ.
In Issue #2, remains to implemented --failed

File: src/assertions.c [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+#define $(x,op)   &&x - &&ADD,
+    IOPS
+#undef $
+  };
+  if (b >= IOPSENUM_COUNT) return a;
+  goto *(&&ADD + array[b]);
+#define $(x,op)   x: return a op c;
+  IOPS
+#undef $
+}
```
Comment (barbu110) [inline src/assertions.c:19]: There is no newline at the end of file.

**Review by barbu110 — APPROVED** [review]

**Pull request merged**